src/connectors/aws_config.py

```diff
@@ -0,0 +1,248 @@
+                        "Condition": {
+                            "StringLike": {
+                                "s3:prefix": [
+                                    f"{prefix}/*"
+                                ]
+                            }
+                        }
+                    }
+                ]
+            }, indent=4),
+        )
+    }
+
+
+def finalize(connection_name):
+    base_name = f'CONFIG_{connection_name}_EVENTS'.upper()
+    pipe = f'data.{base_name}_PIPE'
+    landing_table = f'data.{base_name}_CONNECTION'
+
+    # Step two: Configure the remainder once the role is properly configured.
+    config_ingest_task = f"""
+INSERT INTO {landing_table} (
+  raw, hash_raw, config_capture_time, account_id, aws_region, resource_type, arn, availability_zone, 
+  resource_creation_time, resource_name, resource_Id, relationships, configuration, tags
+)
+SELECT value raw
+    , HASH(value) hash_raw
+    , value:configurationItemCaptureTime::TIMESTAMP_LTZ(9) config_capture_time
+    , value:awsAccountId::STRING account_id
+    , value:awsRegion::STRING aws_region
+    , value:resourceType::STRING aws_region
+    , value:ARN::STRING arn
+    , value:availabilityZone::STRING availability_zone
+    , value:resourceCreationTime::TIMESTAMP_LTZ(9) resource_creation_time
+    , value:resourceName::STRING resource_name
+    , value:resourceId::STRING resource_Id
+    , value:relationships::VARIANT relationships
+    , value:configuration::VARIANT configuration
+    , value:tags::VARIANT tags
+FROM data.{base_name}_STREAM, lateral flatten(input => v:configurationItems)
+WHERE ARRAY_SIZE(v:configurationItems) > 0
+"""
+
+    db.create_stream(
+        name=f'data.{base_name}_STREAM',
+        target=f'data.{base_name}_STAGING'
+    )
+
+    # IAM change takes 5-15 seconds to take effect
+    sleep(5)
+    db.retry(
+        lambda: db.create_pipe(
+            name=pipe,
+            sql=f"COPY INTO data.{base_name}_STAGING(v) FROM @data.{base_name}_STAGE/",
+            replace=True,
+            autoingest=True
+        ),
+        n=10,
+        sleep_seconds_btw_retry=1
+    )
+
+    db.create_task(name=f'data.{base_name}_TASK', schedule='1 minute',
+                   warehouse=WAREHOUSE, sql=config_ingest_task)
+
+    db.execute(f"ALTER PIPE {pipe} REFRESH")
+
+    pipe_description = list(db.fetch(f'DESC PIPE {pipe}'))
+    if len(pipe_description) < 1:
+        return {
+            'newStage': 'error',
+            'newMessage': f"{pipe} does not exist; please reach out to Snowflake Security for assistance."
+        }
+    else:
+        sqs_arn = pipe_description[0]['notification_channel']
+
+    return {
+        'newStage': 'finalized',
+        'newMessage': (
+            f"Please add this SQS Queue ARN to the bucket event notification "
+            f"channel for all object create events: {sqs_arn}"
+        )
+    }
+
+
+def test(base_name):
+    yield db.fetch(f'ls @DATA.{base_name}_STAGE')
```

Comment:
rm or make it adhere to —

    type TestResult = {
        success: bool,
        description: string,
    }

please